htdocs/filefunc.inc.php

```diff
@@ -144,6 +144,7 @@
 $dolibarr_main_document_root = trim($dolibarr_main_document_root);
 $dolibarr_main_document_root_alt = (empty($dolibarr_main_document_root_alt) ? '' : trim($dolibarr_main_document_root_alt));
 
+// TODO incompatibility with socket connection
```

Comment:
Why not replacing 
if !empty()
with
if isset()
Does this work with socket for you ?